### PR TITLE
Configure tests and add support for postgres & solr8

### PR DESCRIPTION
**Pull request by rlskoeser**
- configure CI with a build matrix matching current and future environment
- update test fixtures and a few tests so everything works with postgresql; skip two migration tests that cause truncation errors
- update solr queries to specify edismax for compatibility with solr 8

### CODE REVIEW

File: .github/workflows/unit_tests.yml [hunk 1]
```diff
@@ -72,92 +75,135 @@ jobs:
...
+          pip install -r requirements.txt
+          pip install -r dev-requirements.txt
+
+      # For mysql only, grant permission to the DB_USER to create databases so
+      # that it doesn't fail trying to create the django test database
+      # - name: Grant permissions to mysql user
+      #   if: matrix.db == 'mysql'
+      #   env:
+      #     GRANT_CMD: "grant all privileges on *.* to '${{ env.DB_USER }}'@'%';"
+      #   run: echo "$GRANT_CMD" | mysql -h 127.0.0.1 --user=root --password="$DB_PASSWORD"
```
Comment (thatbudakguy) [inline .github/workflows/unit_tests.yml:163]: no longer needed, i guess?

File: README.rst [hunk 2]
```diff
@@ -123,12 +123,9 @@ server with hot reload using::
 Switching between the webpack dev server and serving from ``static/`` requires a
 restart of your Django dev server to pick up the changed file paths.
 
-
 You must also configure Solr and install the configSet found under ``solr_conf``.
 If a core does not exist, and the configSet is installed correctly, an appropriate
-core will be made for you.
```
Comment (thatbudakguy) [inline README.rst:129]: why does this feel like "If you do not have a lawyer, one will be provided for you"

**Review by thatbudakguy — APPROVED** [review]
everything looks good here. i think we got some black formatting on one file, but no harm in that :)

File: .github/workflows/unit_tests.yml (hunk 1 above)
Comment (rlskoeser) [reply .github/workflows/unit_tests.yml:163]: I couldn't get it to work this way and reverted back to using the mysql root user that the build was using before I added the matrix. I'll go ahead and remove to clean it up

**Comment by rlskoeser** [comment]
> everything looks good here. i think we got some black formatting on one file, but no harm in that :)

whoops, ok — I remember undoing at least one of the formatting things I did out of habit, but I guess I missed one. Need to implement black & isort soon

File: README.rst (hunk 2 above)
Comment (rlskoeser) [reply README.rst:129]: 😆 IDK! but this part will go away in my next PR anyway

**Pull request merged**